fabric/bin/peer.sh

```diff
@@ -433,24 +435,44 @@ handle_lifecycle_chaincode_createenclave() {
         ORDERER_ADDR=$2;
         shift; shift
         ;;
+        -s|--sgx-credentials-path)
+        SGX_CREDENTIALS_PATH=$2
+        ATTESTATION_PARAMS=$(jq -c -n --arg atype "$(cat ${SGX_CREDENTIALS_PATH}/spid_type.txt)" --arg spid "$(cat ${SGX_CREDENTIALS_PATH}/spid.txt)" --arg sig_rl "" '{attestation_type: $atype, hex_spid: $spid, sig_rl: $sig_rl}' | base64 --wrap=0)
+        shift; shift
+        ;;
+        --peerAddresses)
+        PEER_ADDRESS=$2
+        shift; shift
+        ;;
         *)
         die "createenclave: invalid option"
         ;;
         esac
     done
 
+    # peer address must be specified in createenclave
+    [ -z "${PEER_ADDRESS}" ] && die "No peer address specified in createenclave"
+
```

Comment:
should we somehow enforce that `${PEER_ADDRESS}` contains only a single endpoint? Remember it is set like that `--peerAddresses "localhost:7051"`.  I see people already setting multiple endpoints when creating "multiple" enclaves :)

Reply:
> should we somehow enforce that `${PEER_ADDRESS}` contains only a single endpoint? Remember it is set like that `--peerAddresses "localhost:7051"`. I see people already setting multiple endpoints when creating "multiple" enclaves :)

We for sure should check and prevent it in ERCC that there is only a single one, but checking here of course would give much more intuitive feedback.

Reply:
Curiously, the fabric example use one endpoint per `--peerAddresses` instance, though multiple instances of this in the same command.

Anyway, an early check does not hurt.
Added.

Reply:
> Curiously, the fabric example use one endpoint per `--peerAddresses` instance, though multiple instances of this in the same command.

Good point. This made me also realize that you also will have to pass along the (single) `--peerAddresses` field when calling `query ... initEnclave` in `handle_lifecycle_chaincode_initEnclave` or it would fail if that peer is not the default peer ...

Reply:
> This made me also realize that you also will have to pass along the (single) `--peerAddresses` field when calling `query ... initEnclave` in `handle_lifecycle_chaincode_initEnclave` or it would fail if that peer is not the default peer ...

I'll add a concrete clickable suggestion for this ..